src/ucp/core/ucp_context.c

```diff
@@ -1145,6 +1145,8 @@ static ucs_status_t ucp_add_component_resources(ucp_context_h context,
         }
     }
 
+    context->mem_type_mask |= mem_type_mask;
```

Comment:
why |= and not = ?

Reply:
ucp_add_component_resources() is called for each component

Reply:
where is it set to 0?

Reply:
my bad. thought context obj is zeroed on allocation. fixed now.